integration_tests/src/main/python/date_time_test.py

```diff
@@ -185,6 +185,8 @@ def test_to_unix_timestamp_improved(data_gen):
 
 str_date_and_format_gen = [pytest.param(StringGen('[0-9]{4}/[01][0-9]'),'yyyy/MM', marks=pytest.mark.xfail(reason="cudf does no checks")),
         (StringGen('[0-9]{4}/[01][12]/[0-2][1-8]'),'yyyy/MM/dd'),
+        (StringGen('[01][12]/[0-2][1-8]'), 'MM/dd'),
+        (StringGen('[0-2][1-8]/[01][12]'), 'dd/MM'),
```

Comment:
It looks like these tests only generate valid inputs for the formats being tested? If this is the case, it would be good to test against some invalid inputs as well, either here or in the scala `ParseDateTimeSuite` where we have examples of this type of testing.

Reply:
I tested this locally and it looks good. I added the following tests to `ParseDateTimeSuite` to test these new formats against the existing test data.

```scala
testSparkResultsAreEqual("to_unix_timestamp parse dd/MM",
  timestampsAsStrings,
  new SparkConf().set(SQLConf.LEGACY_TIME_PARSER_POLICY.key, "CORRECTED")) {
  df => {
    df.createOrReplaceTempView("df")
    df.sqlContext.sql("SELECT c0, to_unix_timestamp(c0, 'dd/MM') FROM df")
  }
}

testSparkResultsAreEqual("to_unix_timestamp parse MM/dd",
  timestampsAsStrings,
  new SparkConf().set(SQLConf.LEGACY_TIME_PARSER_POLICY.key, "CORRECTED")) {
  df => {
    df.createOrReplaceTempView("df")
    df.sqlContext.sql("SELECT c0, to_unix_timestamp(c0, 'MM/dd') FROM df")
  }
}
```